restart/restart.go

```diff
@@ -0,0 +1,277 @@
+			mapPinPath = filepath.Join(b.HostConfig.BpfMapDefaultPath, models.TCMapPinPath, iface, v)
+		} else if b.Program.ProgType == models.XDPType {
+			mapPinPath = filepath.Join(b.HostConfig.BpfMapDefaultPath, iface, v)
+		}
+		m, err := ebpf.LoadPinnedMap(mapPinPath, nil)
+		if err != nil {
+			return err
+		}
+		(*output).Maps[v] = m
+	}
+	return nil
+}
+
+func getMetricsMaps(input map[string]models.MetaMetricsBPFMap, b *bpfprogs.BPF, conf *config.Config, output *map[string]*bpfprogs.MetricsBPFMap, iface string) error {
+	for k, v := range input {
+		fg := &bpfprogs.MetricsBPFMap{}
+		var pinnedPath string
+		if b.Program.ProgType == models.XDPType {
+			pinnedPath = filepath.Join(b.HostConfig.BpfMapDefaultPath, iface, v.MapName)
+		} else {
+			pinnedPath = filepath.Join(b.HostConfig.BpfMapDefaultPath, models.TCMapPinPath, iface, v.MapName)
+		}
+		m, err := ebpf.LoadPinnedMap(pinnedPath, nil)
+		if err != nil {
+			return err
+		}
+		info, err := m.Info()
+		if err != nil {
+			return err
+		}
+		id, _ := info.ID()
+		fg.BPFMap = bpfprogs.BPFMap{
+			Name:    v.MapName,
+			MapID:   id,
+			Type:    info.Type,
+			BPFProg: b,
+		}
+		fg.Values = ring.New(conf.NMetricSamples)
+		for _, a := range v.Values {
+			fg.Values.Value = a
+			fg.Values = fg.Values.Next()
+		}
+		fg.Aggregator = v.Aggregator
+		fg.Key = v.Key
+		fg.LastValue = v.LastValue
+		(*output)[k] = fg
+		m.Close()
+	}
+	return nil
+}
+
+func DeserilazeProgram(ctx context.Context, r *models.L3AFMetaData, hostconfig *config.Config, iface string) (*bpfprogs.BPF, error) {
+	g := &bpfprogs.BPF{}
+	g.Program = r.Program
+	g.FilePath = r.FilePath
+	g.RestartCount = r.RestartCount
+	g.MapNamePath = r.MapNamePath
+	g.PrevMapNamePath = r.PrevMapNamePath
+	g.PrevProgMapID = ebpf.MapID(r.PrevProgMapID)
+	g.ProgMapID = ebpf.MapID(r.ProgMapID)
+	g.ProgID = ebpf.ProgramID(r.ProgID)
+	g.Ctx = ctx
+	g.HostConfig = hostconfig
+	g.Done = nil
+	g.BpfMaps = make(map[string]bpfprogs.BPFMap)
+	if err := convertBPFMap(r.BpfMaps, g, &g.BpfMaps, iface); err != nil {
+		return nil, err
+	}
+	g.MetricsBpfMaps = make(map[string]*bpfprogs.MetricsBPFMap)
+	if err := getMetricsMaps(r.MetricsBpfMaps, g, hostconfig, &g.MetricsBpfMaps, iface); err != nil {
+		return nil, fmt.Errorf("metrics maps conversion failed")
+	}
+	g.ProgMapCollection = &ebpf.Collection{
+		Programs: make(map[string]*ebpf.Program),
+		Maps:     make(map[string]*ebpf.Map),
+	}
+	if r.XDPLink {
+		linkPinPath := fmt.Sprintf("%s/links/%s/%s_%s", hostconfig.BpfMapDefaultPath, iface, g.Program.Name, g.Program.ProgType)
+		var err error
+		g.XDPLink, err = link.LoadPinnedLink(linkPinPath, nil)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if err := getCollection(r.ProgMapCollection, &g.ProgMapCollection, g, iface); err != nil {
+		return nil, err
+	}
+	return g, nil
+}
+
+func ConvertIfaceMaps(ctx context.Context, input map[string][]*models.L3AFMetaData, output *map[string]*list.List, hostconfig *config.Config) error {
+	return nil
+}
```

Comment:
No use of this method, can we remove, if not using it.

Reply:
Done